src/lib/swapper/manager/utils.ts

```diff
@@ -29,11 +28,9 @@ export const getRatioFromQuote = async (
   */
   const quoteAssets = [quote.sellAsset, quote.buyAsset, feeAsset]
   const usdRatePromises = quoteAssets.map(asset => swapper.getUsdRate(asset))
-  const [sellAssetUsdRate, buyAssetUsdRate, feeAssetUsdRate] = (
-    await Promise.allSettled(usdRatePromises)
-  )
-    .filter(isFulfilled)
-    .map(p => p.value)
+  const [sellAssetUsdRate, buyAssetUsdRate, feeAssetUsdRate] = (await Promise.all(usdRatePromises))
+    .filter(maybeUsdRatePromise => maybeUsdRatePromise.isOk())
+    .map(p => p.unwrap())
```

Comment:
Absolutely! We have `AsyncResultOf` which we could easily wrap for that. Though one issue with this pattern and why it isn't abstracted yet, is that TS loses arity in mapping tuples, so that would only work in specific cases where all promises resolve to the same type

Reply:
Ah sure, I'm guessing the 'ol `as const` doesn't get us out of the woods here, either.